### ISSUE TITLE
Cannot trigger Esc with `jk` after typing whitespace

### ISSUE THREAD

**Issue by stemry83** [VSCodeVim/Vim#5142]
**Describe the bug**
Can't escape to Normal Mode with `jk` with one whitespace in front. Without one whitespace in front I can escape to Normal Mode, no problem, worked as expected.

**To Reproduce**

1.  In Insert Mode
2.  Type one whitespace (with spacebar)
3.  Type `jk`
4.  Literally output `jk` instead of escaping to Normal Mode

**Expected behavior**
Going back to Normal Mode

**Screenshots**
![Typing slowly](https://media.giphy.com/media/LmfYluMrd7bWQJvKFX/giphy.gif)

![Typing normally](https://media.giphy.com/media/RJh0ORi0BXrGufWTMC/giphy.gif)

**Environment:**

- Extension (VsCodeVim) version: v1.16.0
- VSCode version: 1.48.1
- OS: Windows 10 Home (10.0.19041 Build 19041)

**Additional context**
I remapped through this instead of vimrc:
```json
  "vim.insertModeKeyBindings": [
    {
      "before": ["j", "k"],
      "after": ["<Esc>"]
    }]
```
This is especially problematic when typing comment. Any pointer or help is much appreciated! Thank you!

**Comment by berknam** [comment]
I suspect you have `<Space>` as your leader key and probably have other insert mode keybindings set with your leader key. There has been a major overhaul of the remapping logic that should fix that issue. It will be out on the next version.

**Comment by hin-quicksdaq** [comment]
Hello @berknam, thank you for looking into this issue! Really appreciate it!

> I suspect you have <Space> as your leader key and probably have other insert mode keybindings set with your leader key.

I do!

Thanks to your pointer I guess the problem is that whenever vscodevim sees the sequence `<space>jk`, it's interpreted as `space>j`, seeing no such defined trigger so output the literal `<space>jk`. I added the following keybinding and it solves my problem:

```json
  "vim.insertModeKeyBindings": [
    {
      "before": ["<Leader>", "j", "k"],
      "after": ["<Esc>"]
    } 
```
I'll use this quickfix before the next release. Thank you again!

**Comment by stemry83** [comment]
Hello @berknam, thank you for looking into this issue! Really appreciate it!

> I suspect you have as your leader key and probably have other insert mode keybindings set with your leader key.

I do!

Thanks to your pointer I guess the problem is that whenever vscodevim sees the sequence <space>jk, it's interpreted as space>j, seeing no such defined trigger so output the literal <space>jk. I added the following keybinding and it solves my problem:

```json
  "vim.insertModeKeyBindings": [
    {
      "before": ["<Leader>", "j", "k"],
      "after": ["<Esc>"]
    }
]
```
I'll use this quickfix before the next release. Thank you again!

**Comment by berknam** [comment]
@J-Fields @stemry83 This should be fixed on version 1.17.0. Can this one be closed?
